src/main.cpp

```diff
@@ -3756,19 +3756,50 @@ bool static ConnectTip(CValidationState& state, CBlockIndex* pindexNew, CBlock*
     return true;
 }
 
-bool DisconnectBlocksAndReprocess(int blocks)
+bool DisconnectBlocks(int blocks)
 {
     LOCK(cs_main);
 
     CValidationState state;
 
-    LogPrintf("DisconnectBlocksAndReprocess: Got command to replay %d blocks\n", blocks);
+    LogPrintf("%s: Got command to replay %d blocks\n", __func__, blocks);
     for (int i = 0; i <= blocks; i++)
         DisconnectTip(state);
 
     return true;
 }
 
+void ReprocessBlocks(int nBlocks)
+{
+    std::map<uint256, int64_t>::iterator it = mapRejectedBlocks.begin();
+    while (it != mapRejectedBlocks.end()) {
+        //use a window twice as large as is usual for the nBlocks we want to reset
+        if ((*it).second > GetTime() - (nBlocks * Params().TargetSpacing() * 2)) {
+            BlockMap::iterator mi = mapBlockIndex.find((*it).first);
+            if (mi != mapBlockIndex.end() && (*mi).second) {
+                LOCK(cs_main);
+
+                CBlockIndex* pindex = (*mi).second;
+                LogPrintf("%s - %s\n", __func__, (*it).first.ToString());
+
+                CValidationState state;
+                ReconsiderBlock(state, pindex);
+            }
+        }
+        ++it;
+    }
+
+    CValidationState state;
+    {
+        LOCK(cs_main);
+        DisconnectBlocks(nBlocks);
+    }
+
+    if (state.IsValid()) {
```

Comment:
There's something about this that strikes me as odd.  I think it might be because the constructor of CValidationState sets it to valid; so therefore there's no real need to check if it's valid here, since state hasn't been used yet.  Or it's just the strangeness of defining it in the loop above for every call to ReconsiderBlock(), and then again after it's exited the loop.

Can't put my finger on it yet, but the use of CValidationState in this section of code tickles my review sensors.

Reply:
There is no functional change here.
`ReprocessBlocks` has been simply moved from spork.cpp to main.cpp, the implementation is kept intact (except `DisconnectBlocksAndReprocess` which has been renamed `DisconnectBlocks`).

I do agree with you that the use of CValidationState in this section is odd though.
It requires further inspection but (being unrelated to the spork changes, it should be addressed in a separate PR).